### ISSUE TITLE
Edible Furniture

### ISSUE THREAD

**Issue by dinobossytnew** [PluginBugs/Issues-ItemsAdder#2306]
### Terms

- [X] I'm using the very latest version of ItemsAdder and its dependencies.
- [X] I already searched on this [Github page](https://github.com/PluginBugs/Issues-ItemsAdder/issues) to check if the same suggestion was already reported.
- [X] I already searched on the [plugin wiki](https://itemsadder.devs.beer/) to know if the feature is already implemented.
- [X] I already searched on the **Discord** server to know if anyone already has a solution for this or if the feature was already suggested.

### Discord tag (optional)

DinoBoss#6985

### Describe the solution you'd like

Basically create a furniture that when you right click on it you can eat it as if it were a minecraft cake.

### Is your feature request related to a problem?

It could be infinite or there could be bugs but if lone sets it up perfectly it could be a fully AI function.

### Describe alternatives you've considered

the alternative is as I said with IA and MM making an entity that if you place it and right click on it with mm you can eat it, limitations is that I need MM to do it and that I think there could be a problem if I add things to it.

### Additional context

not

**Comment by dinobossytnew** [comment]
furniture replacement

**Comment by LoneDev6** [comment]
Well, this can be implemented with an addon, I think it's a niche feature.

Anyway I'm not really a fan of the vanilla cake eat logic since it's not following the game food logic itself.
It's the only food that can be placed and eaten gradually.


You might be able to implement it with the replace_properties function of ItemsAdder but I didn't test it for this particular scenario.
```yml
info:
  namespace: myitems
items:
  new_cake:
    display_name: Test Cake
    resource:
      generate: false
      material: PAPER
      model_path: item/new_cake
    events:
      placed_itemframe:
        interact:
          replace_properties:
            custom_model_data:
              copy_from_item: new_cake_1
          feed:
            amount: 2
  new_cake_1:
    display_name: Test Cake
    resource:
      generate: false
      material: PAPER
      model_path: item/new_cake
    events:
      placed_itemframe:
        interact:
          replace_properties:
            custom_model_data:
              copy_from_item: new_cake_2
          feed:
            amount: 2
  new_cake_2:
    display_name: Test Cake
    resource:
      generate: false
      material: PAPER
      model_path: item/new_cake
    events:
      placed_itemframe:
        interact:
          replace_properties:
            custom_model_data:
              copy_from_item: new_cake_3
          feed:
            amount: 2
  new_cake_3:
    display_name: Test Cake
    resource:
      generate: false
      material: PAPER
      model_path: item/new_cake
    events:
      placed_itemframe:
        interact:
          feed:
            amount: 2
```
